xml/value.go

```diff
@@ -0,0 +1,257 @@
+// It will auto close the xml element tag.
+func (jv Value) BigDecimal(v *big.Float) {
+	if i, accuracy := v.Int64(); accuracy == big.Exact {
+		jv.Long(i)
+		return
+	}
+
+	jv.openTagWriterFn()
+	jv.w.Write([]byte(v.Text('e', -1)))
+	jv.closeTagWriterFn()
+}
+
+// Null encodes a null element tag like <root></root>.
+// It will auto close the xml element tag.
+func (jv Value) Null() {
+	// write open tag for the parent object
+	jv.openTagWriterFn()
+
+	// close tag
+	jv.closeTagWriterFn()
+}
+
+// Write writes v directly to the xml document
+// if escapeXMLText is set to true, write will escape text.
+// It will auto close the xml element tag.
+func (jv Value) Write(v []byte, escapeXMLText bool) {
+	jv.openTagWriterFn()
+
+	// escape and write xml text
+	if escapeXMLText {
+		escapeText(jv.w, v)
+	} else {
+		// write xml directly
+		jv.w.Write(v)
+	}
+
+	jv.closeTagWriterFn()
+}
+
+// RootElement builds a root element encoding
+func (jv Value) RootElement() (o *Object) {
+	return newObject(jv.w, jv.scratch)
+}
+
+// NestedElement returns a nested element encoding.
+// It returns a point to element object and a close function that will
+// close the element tag.
+func (jv Value) NestedElement() (o *Object, closeFn func()) {
+	// write open tag for the parent object
+	jv.openTagWriterFn()
+
+	return newObject(jv.w, jv.scratch), jv.closeTagWriterFn
+}
+
+// Array returns an array encoder and a close function that will close
+// the array's root element tag. By default, the members of array are
+// wrapped with `<member>` element tag.
+func (jv Value) Array() (a *Array, closeFn func()) {
+	// write open tag for the parent element
+	jv.openTagWriterFn()
+
+	return newArray(jv.w, jv.scratch), jv.closeTagWriterFn
+}
+
+// ArrayWithCustomName returns an array encoder and a close function that will
+// close the array's root element tag.
+//
+// It takes name as an argument, the name will used to wrap xml array entries.
+// for eg, <someList><customName>entry1</customName><someList>
+// Here `customName` element tag will be wrapped on each array member.
+func (jv Value) ArrayWithCustomName(name string) (a *Array, closeFn func()) {
+	// write open tag for the parent element
+	jv.openTagWriterFn()
+
+	return newArrayWithCustomName(jv.w, jv.scratch, name), jv.closeTagWriterFn
+}
+
+// FlattenedArray returns a flattened array encoder. Unlike other array encoders
+// it DOES NOT return an close function.
+//
+// FlattenedArray Encoder wraps each member with array's root element tag, thus
+// flattening the array.
+//
+// for eg,`<someList>entry1</someList><someList>entry2</someList>`.
+func (jv Value) FlattenedArray() (a *Array) {
+	return newFlattenedArray(jv.w, jv.scratch, jv.openTagWriterFn, jv.closeTagWriterFn)
+}
+
+// Map returns a map encoder and a close function that will close
+// the map's root element tag. By default, the map entries are
+// wrapped with `<entry>` element tag.
+func (jv Value) Map() (m *Map, closeFn func()) {
```

Comment:
nit would be helpful to expand this example doc like the flattened map's exaple.